### ISSUE TITLE
I don't know what to do with this error: unexpected error

### ISSUE THREAD

**Issue by mammadsafar** [dart-lang/pub#4319]
```
Information about the latest pub run.

If you believe something is not working right, you can go to
https://github.com/dart-lang/pub/issues/new to post a new issue and attach this file.

Before making this file public, make sure to remove any sensitive information!

Pub version: 3.4.3
Created: 2024-06-26T20:02:30.631342
FLUTTER_ROOT: C:\src\flutter
PUB_HOSTED_URL: <not set>
PUB_CACHE: "<not set>"
Command: dart pub --suppress-analytics --directory '.' get --example
Platform: windows

---- E:\Project\Dart and Flutter\Flutter\code\wordpress_shop\pubspec.yaml ----
name: wordpress_app
description: A new Flutter project.
publish_to: "none"
version: 0.1.0

environment:
  sdk: ">=2.19.6 <3.0.0"

dependencies:
  flutter:
    sdk: flutter

  provider: ^6.0.5
  dio: ^5.1.1
  animated_bottom_navigation_bar: ^1.2.0
  page_transition: ^2.0.9
  collection: ^1.17.0
  intl: 0.19.0
  flutter_inappwebview: ^5.7.2+3
  device_preview: ^1.1.0
  # shared_preferences: ^2.2.3

  # shamsi_date: ^1.0.1
  # persistent_bottom_nav_bar: ^5.0.2
  # flutter_native_splash: ^2.3.11
  # loading_animation_widget: ^1.2.0+4
  # flutter_secure_storage: ^4.2.1

dev_dependencies:
  flutter_test:
    sdk: flutter
  flutter_lints: ^2.0.0

flutter:
  uses-material-design: true

  assets:
    - assets/images/

  fonts:
    - family: iranSans
      fonts:
        - asset: assets/fonts/IRANSans.ttf
    - family: Lalezar
      fonts:
        - asset: assets/fonts/Lalezar.ttf
    - family: YekanBakh
      fonts:
        - asset: assets/fonts/YekanBakh.ttf

---- End pubspec.yaml ----
---- E:\Project\Dart and Flutter\Flutter\code\wordpress_shop\pubspec.lock ----
# Generated by pub
# See https://dart.dev/tools/pub/glossary#lockfile
packages:
  animated_bottom_navigation_bar:
    dependency: "direct main"
    description:
      name: animated_bottom_navigation_bar
      sha256: "2b04a2ae4b0742669e60ddf309467d6a354cefd2d0cd20f4737b1efaf9834cda"
      url: "https://pub.dev"
    source: hosted
    version: "1.3.3"
  ansicolor:
    dependency: transitive
    description:
      name: ansicolor
      sha256: "8bf17a8ff6ea17499e40a2d2542c2f481cd7615760c6d34065cb22bfd22e6880"
      url: "https://pub.dev"
    source: hosted
    version: "2.0.2"
  archive:
    dependency: transitive
    description:
      name: archive
      sha256: cb6a278ef2dbb298455e1a713bda08524a175630ec643a242c399c932a0a1f7d
      url: "https://pub.dev"
    source: hosted
    version: "3.6.1"
  args:
    dependency: transitive
    description:
      name: args
      sha256: "7cf60b9f0cc88203c5a190b4cd62a99feea42759a7fa695010eb5de1c0b2252a"
      url: "https://pub.dev"
    source: hosted
    version: "2.5.0"
  async:
    dependency: transitive
    description:
      name: async
      sha256: "947bfcf187f74dbc5e146c9eb9c0f10c9f8b30743e341481c1e2ed3ecc18c20c"
      url: "https://pub.dev"
    source: hosted
    version: "2.11.0"
  boolean_selector:
    dependency: transitive
    description:
      name: boolean_selector
      sha256: "6cfb5af12253eaf2b368f07bacc5a80d1301a071c73360d746b7f2e32d762c66"
      url: "https://pub.dev"
    source: hosted
    version: "2.1.1"
  characters:
    dependency: transitive
    description:
      name: characters
      sha256: "04a925763edad70e8443c99234dc3328f442e811f1d8fd1a72f1c8ad0f69a605"
      url: "https://pub.dev"
    source: hosted
    version: "1.3.0"
  clock:
    dependency: transitive
    description:
      name: clock
      sha256: cb6d7f03e1de671e34607e909a7213e31d7752be4fb66a86d29fe1eb14bfb5cf
      url: "https://pub.dev"
    source: hosted
    version: "1.1.1"
  collection:
    dependency: "direct main"
    description:
      name: collection
      sha256: ee67cb0715911d28db6bf4af1026078bd6f0128b07a5f66fb2ed94ec6783c09a
      url: "https://pub.dev"
    source: hosted
    version: "1.18.0"
  crypto:
    dependency: transitive
    description:
      name: crypto
      sha256: ff625774173754681d66daaf4a448684fb04b78f902da9cb3d308c19cc5e8bab
      url: "https://pub.dev"
    source: hosted
    version: "3.0.3"
  csslib:
    dependency: transitive
    description:
      name: csslib
      sha256: "706b5707578e0c1b[...]iption:
      name: term_glyph
      sha256: a29248a84fbb7c79282b40b8c72a1209db169a2e0542bce341da992fe1bc7e84
      url: "https://pub.dev"
    source: hosted
    version: "1.2.1"
  test_api:
    dependency: transitive
    description:
      name: test_api
      sha256: "9955ae474176f7ac8ee4e989dadfb411a58c30415bcfb648fa04b2b8a03afa7f"
      url: "https://pub.dev"
    source: hosted
    version: "0.7.0"
  typed_data:
    dependency: transitive
    description:
      name: typed_data
      sha256: facc8d6582f16042dd49f2463ff1bd6e2c9ef9f3d5da3d9b087e244a7b564b3c
      url: "https://pub.dev"
    source: hosted
    version: "1.3.2"
  universal_io:
    dependency: transitive
    description:
      name: universal_io
      sha256: "1722b2dcc462b4b2f3ee7d188dad008b6eb4c40bbd03a3de451d82c78bba9aad"
      url: "https://pub.dev"
    source: hosted
    version: "2.2.2"
  vector_math:
    dependency: transitive
    description:
      name: vector_math
      sha256: "80b3257d1492ce4d091729e3a67a60407d227c27241d6927be0130c98e741803"
      url: "https://pub.dev"
    source: hosted
    version: "2.1.4"
  vm_service:
    dependency: transitive
    description:
      name: vm_service
      sha256: "3923c89304b715fb1eb6423f017651664a03bf5f4b29983627c4da791f74a4ec"
      url: "https://pub.dev"
    source: hosted
    version: "14.2.1"
  web:
    dependency: transitive
    description:
      name: web
      sha256: "97da13628db363c635202ad97068d47c5b8aa555808e7a9411963c533b449b27"
      url: "https://pub.dev"
    source: hosted
    version: "0.5.1"
  win32:
    dependency: transitive
    description:
      name: win32
      sha256: a79dbe579cb51ecd6d30b17e0cae4e0ea15e2c0e66f69ad4198f22a6789e94f4
      url: "https://pub.dev"
    source: hosted
    version: "5.5.1"
  xdg_directories:
    dependency: transitive
    description:
      name: xdg_directories
      sha256: faea9dee56b520b55a566385b84f2e8de55e7496104adada9962e0bd11bcff1d
      url: "https://pub.dev"
    source: hosted
    version: "1.0.4"
  xml:
    dependency: transitive
    description:
      name: xml
      sha256: b015a8ad1c488f66851d762d3090a21c600e479dc75e68328c52774040cf9226
      url: "https://pub.dev"
    source: hosted
    version: "6.5.0"
  yaml:
    dependency: transitive
    description:
      name: yaml
      sha256: "75769501ea3489fca56601ff33454fe45507ea3bfb014161abc3b43ae25989d5"
      url: "https://pub.dev"
    source: hosted
    version: "3.1.2"
sdks:
  dart: ">=3.4.0 <4.0.0"
  flutter: ">=3.22.0"

---- End pubspec.lock ----
---- Log transcript ----
FINE: Pub 3.4.3
MSG : Resolving dependencies...
SLVR: fact: wordpress_app is 0.1.0
SLVR: derived: wordpress_app
SLVR: fact: wordpress_app depends on flutter from sdk
SLVR: fact: wordpress_app depends on provider ^6.0.5
SLVR: fact: wordpress_app depends on dio ^5.1.1
SLVR: fact: wordpress_app depends on animated_bottom_navigation_bar ^1.2.0
SLVR: fact: wordpress_app depends on page_transition ^2.0.9
SLVR: fact: wordpress_app depends on collection ^1.17.0
SLVR: fact: wordpress_app depends on intl 0.19.0
SLVR: fact: wordpress_app depends on flutter_inappwebview ^5.7.2+3
SLVR: fact: wordpress_app depends on device_preview ^1.1.0
SLVR: fact: wordpress_app depends on flutter_test from sdk
SLVR: fact: wordpress_app depends on flutter_lints ^2.0.0
SLVR:   selecting wordpress_app
SLVR:   derived: flutter_lints ^2.0.0
SLVR:   derived: flutter_test from sdk
SLVR:   derived: device_preview ^1.1.0
SLVR:   derived: flutter_inappwebview ^5.7.2+3
SLVR:   derived: intl 0.19.0
SLVR:   derived: collection ^1.17.0
SLVR:   derived: page_transition ^2.0.9
SLVR:   derived: animated_bottom_navigation_bar ^1.2.0
SLVR:   derived: dio ^5.1.1
SLVR:   derived: provider ^6.0.5
SLVR:   derived: flutter from sdk
SLVR:   fact: flutter_lints 2.0.3 depends on lints ^2.0.0
SLVR:     selecting flutter_lints 2.0.3
SLVR:     derived: lints ^2.0.0
SLVR:     fact: flutter_test 0.0.0 from sdk depends on flutter from sdk
SLVR:     fact: flutter_test 0.0.0 from sdk depends on test_api 0.7.0
SLVR:     fact: flutter_test 0.0.0 from sdk depends on matcher 0.12.16+1
SLVR:     fact: flutter_test 0.0.0 from sdk depends on path 1.9.0
SLVR:     fact: flutter_test 0.0.0 from sdk depends on fake_async 1.3.1
SLVR:     fact: flutter_test 0.0.0 from sdk depends on clock 1.1.1
SLVR:     fact: flutter_test 0.0.0 from sdk depends on stack_trace 1.11.1
SLVR:     fact: flutter_test 0.0.0 from sdk depends on vector_math 2.1.4
SLVR:     fact: flutter_test 0.0.0 from sdk depends on leak_tracker_flutter_testing 3.0.3
SLVR:     fact: flutter_test 0.0.0 from sdk depends on async 2.11.0
SLVR:     fact: flutter_test 0.0.0 from sdk depends on boolean_selector 2.1.1
SLVR:     fact: flutter_test 0.0.0 from sdk depends on characters 1.3.0
SLVR:     fact: flutter_test 0.0.0 from sdk depends on collection 1.18.0
SLVR:     fact: flutter_test 0.0.0 from sdk depends on leak_tracker 10.0.4
SLVR:     fact: flutter_test 0.0.0 from sdk depends on leak_tracker_testing 3.0.1
SLVR:     fact: flutter_test 0.0.0 from sdk depends on material_color_utilities 0.8.0
SLVR:     fact: flutter_test 0.0.0 from sdk depends on meta 1.12.0
SLVR:     fact: flutter_test 0.0.0 from sdk depends on source_span 1.10.0
SLVR:     fact: flutter_test 0.0.0 from sdk depends on stream_channel 2.1.2
SLVR:     fact: flutter_test 0.0.0 from sdk depends on string_scanner 1.2.0
SLVR:     fact: flutter_test 0.0.0 from sdk depends on term_glyph 1.2.1
SLVR:     fact: flutter_test 0.0.0 from sdk depends on vm_service 14.2.1
SLVR:       selecting flutter_test 0.0.0 from sdk
SLVR:       derived: vm_service 14.2.1
SLVR:       derived: term_glyph 1.2.1
SLVR:       derived: string_scanner 1.2.0
SLVR:       derived: stream_channel 2.1.2
SLVR:       derived: source_span 1.10.0
SLVR:       derived: meta 1.12.0
SLVR:       derived: material_color_utilities 0.8.0
SLVR:       derived: leak_tracker_testing 3.0.1
SLVR:       derived: leak_tracker 10.0.4
SLVR:       derived: collection 1.18.0
SLVR:       derived: characters 1.3.0
SLVR:       derived: boolean_selector 2.1.1
SLVR:       derived: async 2.11.0
SLVR:       derived: leak_tracker_flutter_testing 3.0.3
SLVR:       derived: vector_math 2.1.4
SLVR:       derived: stack_trace 1.11.1
SLVR:       derived: clock 1.1.1
SLVR:       derived: fake_async 1.3.1
SLVR:       derived: path 1.9.0
SLVR:       derived: matcher 0.12.16+1
SLVR:       derived: test_api 0.7.0
SLVR:       fact: device_preview 1.2.0 depends on flutter from sdk
SLVR:       fact: device_preview 1.2.0 depends on flutter_localizations from sdk
SLVR:       fact: device_preview 1.2.0 depends on provider ^6.1.2
SLVR:       fact: device_preview 1.2.0 depends on device_frame ^1.2.0
SLVR:       fact: device_preview 1.2.0 depends on freezed_annotation ^2.4.1
SLVR:       fact: device_preview 1.2.0 depends on json_annotation ^4.9.0
SLVR:       fact: device_preview 1.2.0 depends on shared_preferences ^2.2.3
SLVR:       fact: device_preview 1.2.0 depends on collection ^1.18.0
SLVR:         selecting device_preview 1.2.0
SLVR:         derived: shared_preferences ^2.2.3
SLVR:         derived: json_annotation ^4.9.0
SLVR:         derived: freezed_annotation ^2.4.1
SLVR:         derived: device_frame ^1.2.0
SLVR:         derived: provider ^6.1.2
SLVR:         derived: flutter_localizations from sdk
SLVR:         fact: flutter_inappwebview 5.8.0 depends on flutter from sdk
SLVR:           selecting flutter_inappwebview 5.8.0
SLVR:           fact: intl 0.19.0 depends on clock ^1.1.0
SLVR:           fact: intl 0.19.0 depends on meta ^1.0.2
SLVR:           fact: intl 0.19.0 depends on path ^1.8.0
SLVR:             selecting intl 0.19.0
SLVR:               selecting collection 1.18.0
SLVR:               fact: page_transition 2.1.0 depends on flutter from sdk
SLVR:                 selecting page_transition 2.1.0
SLVR:                 fact: animated_bottom_navigation_bar 1.3.3 depends on flutter from sdk
SLVR:                   selecting animated_bottom_navigation_bar 1.3.3
SLVR:                   fact: dio 5.4.3+1 depends on async ^2.8.2
SLVR:                   fact: dio 5.4.3+1 depends on http_parser ^4.0.0
SLVR:                   fact: dio 5.4.3+1 depends on meta ^1.5.0
SLVR:                   fact: dio 5.4.3+1 depends on path ^1.8.0
SLVR:                     selecting dio 5.4.3+1
SLVR:                     derived: http_parser ^4.0.0
SLVR:                     fact: provider 6.1.2 depends on collection ^1.15.0
SLVR:                     fact: provider 6.1.2 depends on flutter from sdk
SLVR:                     fact: provider 6.1.2 depends on nested ^1.0.0
SLVR:                       selecting provider 6.1.2
SLVR:                       derived: nested ^1.0.0
SLVR:                       fact: flutter 0.0.0 from sdk depends on characters 1.3.0
SLVR:                       fact: flutter 0.0.0 from sdk depends on collection 1.18.0
SLVR:                       fact: flutter 0.0.0 from sdk depends on material_color_utilities 0.8.0
SLVR:                       fact: flutter 0.0.0 from sdk depends on meta 1.12.0
SLVR:                       fact: flutter 0.0.0 from sdk depends on vector_math 2.1.4
SLVR:                       fact: flutter 0.0.0 from sdk depends on sky_engine from sdk
SLVR:                         selecting flutter 0.0.0 from sdk
SLVR:                         derived: sky_engine from sdk
SLVR:                           selecting lints 2.1.1
SLVR:                             selecting vm_service 14.2.1
SLVR:                               selecting term_glyph 1.2.1
SLVR:                               fact: string_scanner 1.2.0 depends on source_span ^1.8.0
SLVR:                                 selecting string_scanner 1.2.0
SLVR:                                 fact: stream_channel 2.1.2 depends on async ^2.5.0
SLVR:                                   selecting stream_channel 2.1.2
SLVR:                                   fact: source_span 1.10.0 depends on collection ^1.15.0
SLVR:                                   fact: source_span 1.10.0 depends on path ^1.8.0
SLVR:                                   fact: source_span 1.10.0 depends on term_glyph ^1.2.0
SLVR:                                     selecting source_span 1.10.0
SLVR:                                       selecting meta 1.12.0
SLVR:                                       fact: material_color_utilities 0.8.0 depends on collection ^1.15.0
SLVR:                                         selecting material_color_utilities 0.8.0
SLVR:                                         fact: leak_tracker_testing 3.0.1 depends on leak_tracker >=9.0.0 <11.0.0
SLVR:                                         fact: leak_tracker_testing 3.0.1 depends on matcher ^0.12.16
SLVR:                                         fact: leak_tracker_testing 3.0.1 depends on meta ^1.11.0
SLVR:                                           selecting leak_tracker_testing 3.0.1
SLVR:                                           fact: leak_tracker 10.0.4 depends on clock ^1.1.1
SLVR:                                           fact: leak_tracker 10.0.4 depends on collection ^1.15.0
SLVR:                                           fact: leak_tracker 10.0.4 depends on meta ^1.8.0
SLVR:                                           fact: leak_tracker 10.0.4 depends on path ^1.8.3
SLVR:                                           fact: leak_tracker 10.0.4 depends on vm_service >=11.10.0 <15.0.0
SLVR:                                             selecting leak_tracker 10.0.4
SLVR:                                               selecting characters 1.3.0
SLVR:                                               fact: boolean_selector 2.1.1 depends on source_span ^1.8.0
SLVR:                                               fact: boolean_selector 2.1.1 depends on string_scanner ^1.1.0
SLVR:                                                 selecting boolean_selector 2.1.1
SLVR:                                                 fact: async 2.11.0 depends on collection ^1.15.0
SLVR:                                                 fact: async 2.11.0 depends on meta ^1.1.7
SLVR:                                                   selecting async 2.11.0
SLVR:                                                   fact: leak_tracker_flutter_testing 3.0.3 depends on flutter from sdk
SLVR:                                                   fact: leak_tracker_flutter_testing 3.0.3 depends on leak_tracker ^10.0.4
SLVR:                                                   fact: leak_tracker_flutter_testing 3.0.3 depends on leak_tracker_testing ^3.0.1
SLVR:                                                   fact: leak_tracker_flutter_testing 3.0.3 depends on matcher ^0.12.16
SLVR:                                                   fact: leak_tracker_flutter_testing 3.0.3 depends on meta ^1.8.0
SLVR:                                                     selecting leak_tracker_flutter_testing 3.0.3
SLVR:                                                       selecting vector_math 2.1.4
SLVR:                                                       fact: stack_trace 1.11.1 depends on path ^1.8.0
SLVR:                                                         selecting stack_trace 1.11.1
SLVR:                                                           selecting clock 1.1.1
SLVR:                                                           fact: fake_async 1.3.1 depends on clock ^1.1.0
SLVR:                                                           fact: fake_async 1.3.1 depends on collection ^1.15.0
SLVR:                                                             selecting fake_async 1.3.1
SLVR:                                                               selecting path 1.9.0
SLVR:                                                               fact: matcher 0.12.16+1 depends on async ^2.10.0
SLVR:                                                               fact: matcher 0.12.16+1 depends on meta ^1.8.0
SLVR:                                                               fact: matcher 0.12.16+1 depends on stack_trace ^1.10.0
SLVR:                                                               fact: matcher 0.12.16+1 depends on term_glyph ^1.2.0
SLVR:                                                               fact: matcher 0.12.16+1 depends on test_api >=0.5.0 <0.8.0
SLVR:                                                                 selecting matcher 0.12.16+1
SLVR:                                                                 fact: test_api 0.7.0 depends on async ^2.5.0
SLVR:                                                                 fact: test_api 0.7.0 depends on boolean_selector ^2.1.0
SLVR:                                                                 fact: test_api 0.7.0 depends on collection ^1.15.0
SLVR:                                                                 fact: test_api 0.7.0 depends on meta ^1.3.0
SLVR:                                                                 fact: test_api 0.7.0 depends on source_span ^1.8.0
SLVR:                                                                 fact: test_api 0.7.0 depends on stack_trace ^1.10.0
SLVR:                                                                 fact: test_api 0.7.0 depends on stream_channel ^2.1.0
SLVR:                                                                 fact: test_api 0.7.0 depends on string_scanner ^1.1.0
SLVR:                                                                 fact: test_api 0.7.0 depends on term_glyph ^1.2.0
SLVR:                                                                   selecting test_api 0.7.0
SLVR:                                                                   fact: shared_preferences 2.2.3 depends on flutter from sdk
SLVR:                                                                   fact: shared_preferences 2.2.3 depends on shared_preferences_android ^2.1.0
SLVR:                                                                   fact: shared_preferences 2.2.3 depends on shared_preferences_foundation ^2.3.5
SLVR:                                                                   fact: shared_preferences 2.2.3 depends on shared_preferences_linux ^2.2.0
SLVR:                                                                   fact: shared_preferences 2.2.3 depends on shared_preferences_platform_interface ^2.3.0
SLVR:                                                                   fact: shared_preferences 2.2.3 depends on shared_preferences_web ^2.1.0
SLVR:                                                                   fact: shared_preferences 2.2.3 depends on shared_preferences_windows ^2.2.0
SLVR:                                                                     selecting shared_preferences 2.2.3
SLVR:                                                                     derived: shared_preferences_windows ^2.2.0
SLVR:                                                                     derived: shared_preferences_web ^2.1.0
SLVR:                                                                     derived: shared_preferences_platform_interface ^2.3.0
SLVR:                                                                     derived: shared_preferences_linux ^2.2.0
SLVR:                                                                     derived: shared_preferences_foundation ^2.3.5
SLVR:                                                                     derived: shared_preferences_android ^2.1.0
SLVR:                                                                     fact: json_annotation 4.9.0 depends on meta ^1.4.0
SLVR:                                                                       selecting json_annotation 4.9.0
SLVR:                                                                       fact: freezed_annotation 2.4.1 depends on collection ^1.15.0
SLVR:                                                                       fact: freezed_annotation 2.4.1 depends on json_annotation ^4.6.0
SLVR:                                                                       fact: freezed_annotation 2.4.1 depends on meta ^1.7.0
SLVR:                                                                         selecting freezed_annotation 2.4.1
SLVR:                                                                         fact: device_frame 1.2.0 depends on freezed_annotation ^2.0.3
SLVR:                                                                         fact: device_frame 1.2.0 depends on flutter from sdk
SLVR:                                                                           selecting device_frame 1.2.0
SLVR:                                                                           fact: flutter_localizations 0.0.0 from sdk depends on flutter from sdk
SLVR:                                                                           fact: flutter_localizations 0.0.0 from sdk depends on intl 0.19.0
SLVR:                                                                           fact: flutter_localizations 0.0.0 from sdk depends on characters 1.3.0
SLVR:                                                                           fact: flutter_localizations 0.0.0 from sdk depends on clock 1.1.1
SLVR:                                                                           fact: flutter_localizations 0.0.0 from sdk depends on collection 1.18.0
SLVR:                                                                           fact: flutter_localizations 0.0.0 from sdk depends on material_color_utilities 0.8.0
SLVR:                                                                           fact: flutter_localizations 0.0.0 from sdk depends on meta 1.12.0
SLVR:                                                                           fact: flutter_localizations 0.0.0 from sdk depends on path 1.9.0
SLVR:                                                                           fact: flutter_localizations 0.0.0 from sdk depends on vector_math 2.1.4
SLVR:                                                                             selecting flutter_localizations 0.0.0 from sdk
SLVR:                                                                             fact: http_parser 4.0.2 depends on collection ^1.15.0
SLVR:                                                                             fact: http_parser 4.0.2 depends on source_span ^1.8.0
SLVR:                                                                             fact: http_parser 4.0.2 depends on string_scanner ^1.1.0
SLVR:                                                                             fact: http_parser 4.0.2 depends on typed_data ^1.3.0
SLVR:                                                                               selecting http_parser 4.0.2
SLVR:                                                                               derived: typed_data ^1.3.0
SLVR:                                                                               fact: nested 1.0.0 depends on flutter from sdk
SLVR:                                                                                 selecting nested 1.0.0
SLVR:                                                                                   selecting sky_engine 0.0.99 from sdk
SLVR:                                                                                   fact: shared_preferences_windows 2.3.2 depends on file >=6.0.0 <8.0.0
SLVR:                                                                                   fact: shared_preferences_windows 2.3.2 depends on flutter from sdk
SLVR:                                                                                   fact: shared_preferences_windows 2.3.2 depends on path ^1.8.0
SLVR:                                                                                   fact: shared_preferences_windows 2.3.2 depends on path_provider_platform_interface ^2.0.0
SLVR:                                                                                   fact: shared_preferences_windows 2.3.2 depends on path_provider_windows ^2.0.0
SLVR:                                                                                   fact: shared_preferences_windows 2.3.2 depends on shared_preferences_platform_interface ^2.3.0
SLVR:                                                                                     selecting shared_preferences_windows 2.3.2
SLVR:                                                                                     derived: path_provider_windows ^2.0.0
SLVR:                                                                                     derived: path_provider_platform_interface ^2.0.0
SLVR:                                                                                     derived: file >=6.0.0 <8.0.0
SLVR:                                                                                     fact: shared_preferences_web 2.3.0 depends on flutter from sdk
SLVR:                                                                                     fact: shared_preferences_web 2.3.0 depends on flutter_web_plugins from sdk
SLVR:                                                                                     fact: shared_preferences_web 2.3.0 depends on shared_preferences_platform_interface ^2.3.0
SLVR:                                                                                     fact: shared_preferences_web 2.3.0 depends on web ^0.5.0
SLVR:                                                                                       selecting shared_preferences_web 2.3.0
SLVR:                                                                                       derived: web ^0.5.0
SLVR:                                                                                       derived: flutter_web_plugins from sdk
SLVR:                                                                                       fact: shared_preferences_platform_interface 2.3.2 depends on flutter from sdk
SLVR:                                                                                       fact: shared_preferences_platform_interface 2.3.2 depends on plugin_platform_interface ^2.1.7
SLVR:                                                                                         selecting shared_preferences_platform_interface 2.3.2
SLVR:                                                                                         derived: plugin_platform_interface ^2.1.7
SLVR:                                                                                         fact: shared_preferences_linux 2.3.2 depends on file >=6.0.0 <8.0.0
SLVR:                                                                                         fact: shared_preferences_linux 2.3.2 depends on flutter from sdk
SLVR:                                                                                         fact: shared_preferences_linux 2.3.2 depends on path ^1.8.0
SLVR:                                                                                         fact: shared_preferences_linux 2.3.2 depends on path_provider_linux ^2.0.0
SLVR:                                                                                         fact: shared_preferences_linux 2.3.2 depends on path_provider_platform_interface ^2.0.0
SLVR:                                                                                         fact: shared_preferences_linux 2.3.2 depends on shared_preferences_platform_interface ^2.3.0
SLVR:                                                                                           selecting shared_preferences_linux 2.3.2
SLVR:                                                                                           derived: path_provider_linux ^2.0.0
SLVR:                                                                                           fact: shared_preferences_foundation 2.4.0 depends on flutter from sdk
SLVR:                                                                                           fact: shared_preferences_foundation 2.4.0 depends on shared_preferences_platform_interface ^2.3.0
SLVR:                                                                                             selecting shared_preferences_foundation 2.4.0
SLVR:                                                                                             fact: shared_preferences_android 2.2.3 depends on flutter from sdk
SLVR:                                                                                             fact: shared_preferences_android 2.2.3 depends on shared_preferences_platform_interface ^2.3.0
SLVR:                                                                                               selecting shared_preferences_android 2.2.3
SLVR:                                                                                               fact: typed_data 1.3.2 depends on collection ^1.15.0
SLVR:                                                                                                 selecting typed_data 1.3.2
SLVR:                                                                                                 fact: path_provider_windows 2.2.1 depends on ffi ^2.0.0
SLVR:                                                                                                 fact: path_provider_windows 2.2.1 depends on flutter from sdk
SLVR:                                                                                                 fact: path_provider_windows 2.2.1 depends on path ^1.8.0
SLVR:                                                                                                 fact: path_provider_windows 2.2.1 depends on path_provider_platform_interface ^2.1.0
SLVR:                                                                                                 fact: path_provider_windows 2.2.1 depends on win32 >=2.1.0 <6.0.0
SLVR:                                                                                                   selecting path_provider_windows 2.2.1
SLVR:                                                                                                   derived: win32 >=2.1.0 <6.0.0
SLVR:                                                                                                   derived: path_provider_platform_interface ^2.1.0
SLVR:                                                                                                   derived: ffi ^2.0.0
SLVR:                                                                                                   fact: path_provider_platform_interface 2.1.2 depends on flutter from sdk
SLVR:                                                                                                   fact: path_provider_platform_interface 2.1.2 depends on platform ^3.0.0
SLVR:                                                                                                   fact: path_provider_platform_interface 2.1.2 depends on plugin_platform_interface ^2.1.7
SLVR:                                                                                                     selecting path_provider_platform_interface 2.1.2
SLVR:                                                                                                     derived: platform ^3.0.0
SLVR:                                                                                                     fact: file 7.0.0 depends on meta ^1.9.1
SLVR:                                                                                                     fact: file 7.0.0 depends on path ^1.8.3
SLVR:                                                                                                       selecting file 7.0.0
SLVR:                                                                                                         selecting web 0.5.1
SLVR:                                                                                                         fact: flutter_web_plugins 0.0.0 from sdk depends on flutter from sdk
SLVR:                                                                                                         fact: flutter_web_plugins 0.0.0 from sdk depends on characters 1.3.0
SLVR:                                                                                                         fact: flutter_web_plugins 0.0.0 from sdk depends on collection 1.18.0
SLVR:                                                                                                         fact: flutter_web_plugins 0.0.0 from sdk depends on material_color_utilities 0.8.0
SLVR:                                                                                                         fact: flutter_web_plugins 0.0.0 from sdk depends on meta 1.12.0
SLVR:                                                                                                         fact: flutter_web_plugins 0.0.0 from sdk depends on vector_math 2.1.4
SLVR:                                                                                                           selecting flutter_web_plugins 0.0.0 from sdk
SLVR:                                                                                                           fact: plugin_platform_interface 2.1.8 depends on meta ^1.3.0
SLVR:                                                                                                             selecting plugin_platform_interface 2.1.8
SLVR:                                                                                                             fact: path_provider_linux 2.2.1 depends on ffi >=1.1.2 <3.0.0
SLVR:                                                                                                             fact: path_provider_linux 2.2.1 depends on flutter from sdk
SLVR:                                                                                                             fact: path_provider_linux 2.2.1 depends on path ^1.8.0
SLVR:                                                                                                             fact: path_provider_linux 2.2.1 depends on path_provider_platform_interface ^2.1.0
SLVR:                                                                                                             fact: path_provider_linux 2.2.1 depends on xdg_directories >=0.2.0 <2.0.0
SLVR:                                                                                                               selecting path_provider_linux 2.2.1
SLVR:                                                                                                               derived: xdg_directories >=0.2.0 <2.0.0
SLVR:                                                                                                               fact: win32 5.5.1 depends on ffi ^2.1.2
SLVR:                                                                                                                 selecting win32 5.5.1
SLVR:                                                                                                                 derived: ffi ^2.1.2
SLVR:                                                                                                                   selecting ffi 2.1.2
SLVR:                                                                                                                     selecting platform 3.1.5
SLVR:                                                                                                                     fact: xdg_directories 1.0.4 depends on meta ^1.3.0
SLVR:                                                                                                                     fact: xdg_directories 1.0.4 depends on path ^1.8.0
SLVR:                                                                                                                       selecting xdg_directories 1.0.4
IO  : Get versions from https://pub.dev/api/packages/flutter_lints.
IO  : HTTP GET https://pub.dev/api/packages/flutter_lints
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/flutter_lints
    | took 0:00:00.628613
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 321
    | referrer-policy: no-referrer
IO  : Get versions from https://pub.dev/api/packages/lints.
IO  : HTTP GET https://pub.dev/api/packages/lints
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/lints
    | took 0:00:00.484137
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 313
    | referrer-policy: no-referrer
IO  : Get versions from https://pub.dev/api/packages/vm_service.
IO  : HTTP GET https://pub.dev/api/packages/vm_service
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/vm_service
    | took 0:00:00.480129
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 318
    | referrer-policy: no-referrer
IO  : Get versions from https://pub.dev/api/packages/stream_channel.
IO  : HTTP GET https://pub.dev/api/packages/stream_channel
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/stream_channel
    | took 0:00:00.500573
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 322
    | referrer-policy: no-referrer
IO  : Get versions from https://pub.dev/api/packages/material_color_utilities.
IO  : HTTP GET https://pub.dev/api/packages/material_color_utilities
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/material_color_utilities
    | took 0:00:00.484817
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 332
    | referrer-policy: no-referrer
IO  : Get versions from https://pub.dev/api/packages/leak_tracker_testing.
IO  : HTTP GET https://pub.dev/api/packages/leak_tracker_testing
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/leak_tracker_testing
    | took 0:00:00.512121
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 328
    | referrer-policy: no-referrer
IO  : Get versions from https://pub.dev/api/packages/leak_tracker.
IO  : HTTP GET https://pub.dev/api/packages/leak_tracker
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/leak_tracker
    | took 0:00:00.492616
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 320
    | referrer-policy: no-referrer
IO  : Get versions from https://pub.dev/api/packages/characters.
IO  : HTTP GET https://pub.dev/api/packages/characters
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/characters
    | took 0:00:00.499956
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 318
    | referrer-policy: no-referrer
IO  : Get versions from https://pub.dev/api/packages/boolean_selector.
IO  : HTTP GET https://pub.dev/api/packages/boolean_selector
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/boolean_selector
    | took 0:00:00.496824
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 324
    | referrer-policy: no-referrer
IO  : Get versions from https://pub.dev/api/packages/leak_tracker_flutter_testing.
IO  : HTTP GET https://pub.dev/api/packages/leak_tracker_flutter_testing
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/leak_tracker_flutter_testing
    | took 0:00:00.491736
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 336
    | referrer-policy: no-referrer
IO  : Get versions from https://pub.dev/api/packages/vector_math.
IO  : HTTP GET https://pub.dev/api/packages/vector_math
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/vector_math
    | took 0:00:00.494022
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 319
    | referrer-policy: no-referrer
IO  : Get versions from https://pub.dev/api/packages/stack_trace.
IO  : HTTP GET https://pub.dev/api/packages/stack_trace
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/stack_trace
    | took 0:00:00.477064
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 319
    | referrer-policy: no-referrer
IO  : Get versions from https://pub.dev/api/packages/fake_async.
IO  : HTTP GET https://pub.dev/api/packages/fake_async
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/fake_async
    | took 0:00:00.449978
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 318
    | referrer-policy: no-referrer
IO  : Get versions from https://pub.dev/api/packages/matcher.
IO  : HTTP GET https://pub.dev/api/packages/matcher
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/matcher
    | took 0:00:00.472788
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 315
    | referrer-policy: no-referrer
IO  : Get versions from https://pub.dev/api/packages/test_api.
IO  : HTTP GET https://pub.dev/api/packages/test_api
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/test_api
    | took 0:00:00.478430
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 316
    | referrer-policy: no-referrer
SLVR: Version solving took 0:00:07.958115 seconds.
    | Tried 1 solutions.
FINE: Resolving dependencies finished (8.0s).
MSG : Downloading packages...
FINE: Downloading packages finished (0.0s).
IO  : Get versions from https://pub.dev/api/packages/dio.
IO  : HTTP GET https://pub.dev/api/packages/dio
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: windows
    | X-Pub-Command: get
    | X-Pub-Session-ID: B28EB89B-61C0-4282-9DF9-A3D7C328CE4D
    | X-Pub-Environment: vscode.dart-code:flutter_cli:verify:run
    | user-agent: Dart pub 3.4.3
IO  : HTTP response 403 Forbidden for GET https://pub.dev/api/packages/dio
    | took 0:00:00.490339
    | content-type: text/html; charset=UTF-8
    | alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
    | content-length: 311
    | referrer-policy: no-referrer
ERR : Null check operator used on a null value
FINE: Exception type: _TypeError
ERR : package:pub/src/source/hosted.dart 760:33              HostedSource._getAdvisories.readAdvisoriesFromCache
    | ===== asynchronous gap ===========================
    | package:pub/src/source/hosted.dart 778:12              HostedSource._getAdvisories
    | ===== asynchronous gap ===========================
    | package:pub/src/source/hosted.dart 1066:24             HostedSource.getAdvisoriesForPackageVersion
    | ===== asynchronous gap ===========================
    | package:pub/src/solver/report.dart 414:26              SolveReport._reportPackage
    | ===== asynchronous gap ===========================
    | package:pub/src/solver/report.dart 218:18              SolveReport._reportChanges
    | ===== asynchronous gap ===========================
    | package:pub/src/solver/report.dart 73:21               SolveReport.show
    | ===== asynchronous gap ===========================
    | package:pub/src/entrypoint.dart 568:5                  Entrypoint.acquireDependencies
    | ===== asynchronous gap ===========================
    | package:pub/src/command/get.dart 77:5                  GetCommand.runProtected
    | ===== asynchronous gap ===========================
    | package:pub/src/command.dart 196:7                     PubCommand.run
    | ===== asynchronous gap ===========================
    | package:args/command_runner.dart 212:13                CommandRunner.runCommand
    | ===== asynchronous gap ===========================
    | package:dartdev/dartdev.dart 232:18                    DartdevRunner.runCommand
    | ===== asynchronous gap ===========================
    | package:dartdev/dartdev.dart 46:16                     runDartdev
    | ===== asynchronous gap ===========================
    | C:\b\s\w\ir\x\w\sdk\pkg\dartdev\bin\dartdev.dart 13:5  main
ERR : This is an unexpected error. The full log and other details are collected in:
    | 
    |     C:\Users\Mammadsafar\AppData\Local\Pub\Cache\log\pub_log.txt
    | 
    | Consider creating an issue on https://github.com/dart-lang/pub/issues/new
    | and attaching the relevant parts of that log file.
---- End log transcript ----
```

**Comment by sigurdm** [comment]
What dart sdk version are you on?

I believe this is a duplicate of https://github.com/dart-lang/pub/issues/4269

This should be fixed in 3.4.4 https://github.com/dart-lang/sdk/blob/stable/CHANGELOG.md#344---2024-06-12
